Scripts/Python/xSimpleImagerClueHandler.py

```diff
@@ -0,0 +1,109 @@
+(or a modified version of those libraries),
+containing parts covered by the terms of the Bink SDK EULA, 3ds Max EULA,
+PhysX SDK EULA, DirectX SDK EULA, OpenSSL and SSLeay licenses, IJG
+JPEG Library README, Windows Media SDK EULA, or QuickTime SDK EULA, the
+licensors of this Program grant you additional
+permission to convey the resulting work. Corresponding Source for a
+non-source form of such a combination shall include the source code for
+the parts of OpenSSL and IJG JPEG Library used as well as that of the covered
+work.
+
+You can contact Cyan Worlds, Inc. by email legal@cyan.com
+ or by snail mail at:
+      Cyan Worlds, Inc.
+      14617 N Newport Hwy
+      Mead, WA   99021
+
+ *==LICENSE==* """
+
+from Plasma import *
+from PlasmaTypes import *
+
+import random
+
+class xSimpleImagerClueHandler:
+
+    def __init__(self, baseImager: ptSceneobject, imagerMaxSize: int, clueImager: ptSceneobject, clueTime: int, randomMaxTimeOffset: int):
+        PtDebugPrint(f"xSimpleImagerClueHandler: init for imager with max size {imagerMaxSize}, clue time of {clueTime} seconds, and {randomMaxTimeOffset} possible offset", level=kWarningLevel)
+
+        random.seed()
+        self.isShowing = False
+        self.shouldShow = False
+        self.baseImager = baseImager
+        self.imagerMaxSize = imagerMaxSize
+        self.clueImager = clueImager
+        self.clueTime = clueTime
+        self.randomMaxTimeOffset = randomMaxTimeOffset
+
+        # initialize hidden
+        self.clueImager.draw.disable()
+        self.clueImager.physics.suppress(True)
+
+        self.SetClueAlarm()
+
+    def SetClueAlarm(self):
+        randomizedClueTime = self.clueTime + random.randint(0, self.randomMaxTimeOffset)
+        PtDebugPrint(f"xSimpleImagerClueHandler.SetClueAlarm: Set for {randomizedClueTime} seconds", level=kWarningLevel)
+        PtSetAlarm(randomizedClueTime, self, 1)
+
+    def UpdateClueState(self, forceOff=False):
+
+        if self.isShowing or forceOff:
+            # only the owner of the imager changes the images
+            if self.baseImager.isLocallyOwned():
+                # if clue currently being shown, turn it off and flip the regularly scheduled imager back on
+                PtDebugPrint("xSimpleImagerClueHandler.UpdateClueState: Turning the clue off now...", level=kWarningLevel)
+                self.clueImager.draw.disable()
+                self.clueImager.physics.suppress(True)
+                self.baseImager.draw.enable()
+                self.baseImager.physics.suppress(False)
+
+            # set alarm to show the clue again if it was on before we turned it off (versus forcing while already off)
+            # otherwise multiple alarms will be active at the same time
+            if self.isShowing:
+                self.shouldShow = False
+                self.SetClueAlarm()
+
+            self.isShowing = False
+        elif self.shouldShow:
+            # only the owner of the imager changes the images
```

Comment:
Be sure to net force the changes. Otherwise, this action will only happen on the owner's local client.

Reply:
Sorry, I missed this comment from a couple weeks ago. Which changes are you referring to? The enable/disable/suppress things below?

Reply:
Yes and those found above. I can't remember if those changes are already net-forced from the xSI end of things, so please check.

Reply:
This should be resolved when @Hazado pushes to this PR with my updates.